### ISSUE TITLE
Descriptor: when using a variable descriptor count a false positive error is always thrown 

### ISSUE THREAD

**Issue by Luke-Filewalker** [KhronosGroup/Vulkan-ValidationLayers#2501]
**Describe the Issue**
When a descriptor is created with VK_DESCRIPTOR_BINDING_VARIABLE_DESCRIPTOR_COUNT_BIT_EXT set, the following error is always thrown and the error message always says that the variable descriptor size is 0. This happens when the set describtor count is smaller and even when it is equal to the max set describtor count.

**Valid Usage ID**
Validation Error: [ VUID-VkWriteDescriptorSet-dstArrayElement-00321 ] Object 0: handle = 0x887ecb000000004e, type = VK_OBJECT_TYPE_DESCRIPTOR_SET; | MessageID = 0x90f9e073 | vkUpdateDescriptorSets() failed write update validation for VkDescriptorSet 0x887ecb000000004e[] with error: Attempting write update to VkDescriptorSet 0x887ecb000000004e[] allocated with VkDescriptorSetLayout 0xa9ea89000000004c[] binding index #0 array element 0 with 2 writes but variable descriptor size is 0. The Vulkan spec states: The sum of dstArrayElement and descriptorCount must be less than or equal to the number of array elements in the descriptor set binding specified by dstBinding, and all applicable consecutive bindings, as described by consecutive binding updates (https://vulkan.lunarg.com/doc/view/1.2.162.1/windows/1.2-extensions/vkspec.html#VUID-VkWriteDescriptorSet-dstArrayElement-00321)

**Environment:**
 - OS: Windows 10 2004 
 - GPU: Nvidia 2080ti
 - SDK or header version if building from repo: SDK 1.2.162.1
 - Options enabled (synchronization, best practices, etc.):

**Additional context**

**Comment by eliasnaur** [comment]
Did you specify the variable descriptor count through a [vkDescriptorSetVariableDescriptorCountAllocateInfo](https://www.khronos.org/registry/vulkan/specs/1.2-extensions/man/html/VkDescriptorSetVariableDescriptorCountAllocateInfo.html) to vkAllocateDescriptorSets? I got the tip from https://gist.github.com/NotAPenguin0/284461ecc81267fa41a7fbc472cd3afe#creating-the-descriptor-sets.

I got a similar validation error even though variable sized sampler2D arrays worked fine on my setup. However, variable sized image2D arrays *didn't* work until I added the above info, which also fixed the validation error.
